### DISCUSSION TITLE
[Plugin]: Can't upload file to repo- get 422 error.

### DISCUSSION THREAD

**Discussion by BlingDan** [ObsidianPublisher/obsidian-github-publisher#332] [Q&A — Plugin]
Hi~

I've recently been using this plugin to push local files to a github repository and have run into some issues.

I already checked my repository and it's valid and used `Test connection` button, it’s result is `** seems to be valid`.

I already tried to send an example file to the repo , but it hos no reply for this action. 

here is my contnet:

![image-20240413100216884](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413100216884.png)

I checked the console and here is the log

![image-20240413100259782](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413100259782.png)

- Command used : 
  - [ ] Upload single current active note
- here is my settings : 

```json
{
  "github": {
    "branch": "main",
    "automaticallyMergePR": true,
    "dryRun": {
      "enable": false,
      "folderName": "github-publisher"
    },
    "tokenPath": "%configDir%/plugins/%pluginID%/env",
    "api": {
      "tiersForApi": "Github Free/Pro/Team (default)",
      "hostname": ""
    },
    "workflow": {
      "commitMessage": "[PUBLISHER] Merge",
      "name": ""
    },
    "verifiedRepo": true
  },
  "upload": {
    "behavior": "fixed",
    "defaultName": "",
    "rootFolder": "",
    "yamlFolderKey": "",
    "frontmatterTitle": {
      "enable": false,
      "key": "title"
    },
    "replaceTitle": [],
    "replacePath": [],
    "autoclean": {
      "enable": false,
      "excluded": []
    },
    "folderNote": {
      "enable": false,
      "rename": "index.md",
      "addTitle": {
        "enable": false,
        "key": "title"
      }
    },
    "metadataExtractorPath": ""
  },
  "conversion": {
    "hardbreak": false,
    "dataview": true,
    "censorText": [],
    "tags": {
      "inline": false,
      "exclude": [],
      "fields": []
    },
    "links": {
      "internal": false,
      "unshared": false,
      "wiki": false,
      "slugify": "disable"
    }
  },
  "embed": {
    "attachments": true,
    "overrideAttachments": [],
    "keySendFile": [],
    "notes": false,
    "folder": "",
    "convertEmbedToLinks": "keep",
    "charConvert": "->",
    "useObsidianFolder": true
  },
  "plugin": {
    "shareKey": "share",
    "excludedFolder": [],
    "copyLink": {
      "enable": false,
      "links": "",
      "removePart": [],
      "transform": {
        "toUri": true,
        "slugify": "lower",
        "applyRegex": []
      }
    },
    "setFrontmatterKey": "Set"
  }
}
```

I'd appreciate it very much if anyone could do my a favor.

**Comment by Mara-Li** [comment]
Delete the branch created by the plugin on the repo

**Reply by BlingDan** [reply]
thanks for your reply, I delete all branchs except `main` and test connection . it seems to be valid.



![image-20240413131005323](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413131005323.png)

![image-20240413131020612](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413131020612.png)



but when I push my file to repo, it has no answer. I check the console and find it create a new branch. I don’t know why it don’t push file to `main` branch. The new branch also don’t have my pushed file.

here is the log and github branch.

![image-20240413132110170](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413132110170.png)
![image-20240413131919292](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413131919292.png)

**Comment by Mara-Li** [comment]
Have you something in the connexion tab ? The 422 error is pretty large and I think it's more like "detection of another existing branch"…
Maybe you have some security that prevents push ?

**Reply by BlingDan** [reply]
Could you tell me where is the `connexion tab`

I generate a new github token and give it all access like this 

![image-20240413194134841](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413194134841.png)

then modify the `Main branch` to `Zacary-4-13-2024` . But it still don’t work.

![image-20240413194302579](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413194302579.png)

Here is the log, It's the same as what I mentioned at the beginning.😥

![image-20240413194708405](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240413194708405.png)

**Comment by Mara-Li** [comment]
Sorry for the late reply. Could you click on the arrow right after the "plugin:…." ?

**Reply by BlingDan** [reply]
Thanks for your reply.

this is the connexion tab

![image-20240414220801794](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240414220801794.png)

![image-20240414220812534](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240414220812534.png)

![image-20240414220830187](https://raw.githubusercontent.com/BlingDan/ImageHosting/main/img/image-20240414220830187.png)

**Comment by Mara-Li** [comment]
Little question, after the branch creation, have you something about file published/error…? Do you have some securities from github protecting push on branch ?
I checked around and It can cause error

**Reply by BlingDan** [reply]
No,  there is no red error box in the upper right corner. I can only check the message from the console.
The new created branch also don't have  file that I want to push.

**Reply by Mara-Li** [reply]
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/30244939/83d7905c-230b-486d-8209-223cb66557e9)

**Reply by BlingDan** [reply]
ok, thank you for your assistance🥰
